clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -101,6 +101,11 @@
 
     private int assignmentId = 0;
 
+    @Override
+    public String toString() {
+        return subscribedPattern != null ? subscribedPattern.toString() : String.join(",", subscription);
```

Comment:
Since this is a general `toString()`, we probably should handle manual assignment as well.